nbextensions/bulkImportCell/bulkImportCellStates.js

```diff
@@ -40,8 +61,346 @@ define([], () => {
+                        },
+                        jobStatus: {
+                            enabled: true,
+                            visible: true
+                        },
+                        results: {
+                            enabled: false,
+                            visible: true
+                        },
+                        error: {
+                            enabled: false,
+                            visible: false
+                        }
+                    }
+                },
+                action: {
+                    name: 'cancel',
+                    disabled: false
+                }
+            }
+
+        },
+        // apps are running, none are complete yet
+        running: {
+            ui: {
+                tab: {
+                    tabs: {
+                        configure: {
+                            enabled: false,
+                            visible: false,
+                        },
+                        viewConfigure: {
+                            enabled: true,
+                            visible: true
+                        },
+                        info: {
+                            enabled: true,
+                            visible: true
+                        },
+                        jobStatus: {
+                            enabled: true,
+                            visible: true
+                        },
+                        results: {
+                            enabled: false,
+                            visible: true
+                        },
+                        error: {
+                            enabled: false,
+                            visible: false
+                        }
+                    }
+                },
+                action: {
+                    name: 'cancel',
+                    disabled: false
+                }
+            }
+
+        },
+        // at least one child job is complete
+        appPartialComplete: {
+            ui: {
+                tab: {
+                    tabs: {
+                        configure: {
+                            enabled: false,
+                            visible: false,
+                        },
+                        viewConfigure: {
+                            enabled: true,
+                            visible: true
+                        },
+                        info: {
+                            enabled: true,
+                            visible: true
+                        },
+                        jobStatus: {
+                            enabled: true,
+                            visible: true
+                        },
+                        results: {
+                            enabled: true,
+                            visible: true
+                        },
+                        error: {
+                            enabled: false,
+                            visible: false
+                        }
```

Comment:
This configuration pops up numerous times -- is it worth saving it as a variable (`defaultRunConfig` or something) or using a function to generate it? There are also several cases above where the config is this, with one value changed to `false`; could it be created by copying the defaultRunConfig and altering whichever value is relevant?

Reply:
Sure, I'll add that.

Reply:
@ialarmedalien ok, I turned that from a giant declared JSON structure to something assembled from functions.